File: sdks/go/tests/conformance/basic_logging_events_test.go
package conformance

import (
	"context"
	"testing"

	"github.com/astraive/loza/sdks/go"
)

func TestBasicLoggingAndEventFacades(t *testing.T) {
	sink, store := loza.MemorySink()
	if err := loza.Configure(loza.Test().WithSink(sink)); err != nil {
		t.Fatalf("configure: %v", err)
	}

	loza.Notice("notice event", loza.String("family", "logs"))
	loza.Track("checkout.page_view", loza.String("page", "/checkout"))
	loza.Audit("user.login", loza.UserID("u_123"))
	loza.Security("auth.failure", loza.ErrorCode("AUTH_BAD_PASSWORD"))
	loza.Metric("latency", 12.5, loza.String("unit", "ms"))
	loza.Count("requests", 4)
	loza.Gauge("cpu", 0.72)
	loza.Histogram("payload.bytes", 512)
	loza.Breadcrumb("nav.click", loza.String("button", "submit"))

	if err := loza.Flush(context.Background()); err != nil {
		t.Fatalf("flush: %v", err)
	}
	if store.Len() < 9 {
		t.Fatalf("expected 9 emitted facade events, got %d", store.Len())
	}
}

File: sdks/js/tests/basic_logging_events.test.ts
import { afterEach, it } from 'node:test';
import assert from 'node:assert/strict';
import * as loza from '../src/index.ts';

afterEach(() => {
  loza.reset();
});

it('covers basic logging and event facades', async () => {
  const sink = new loza.MemorySink();
  loza.configure(loza.production('catalog').withSink(sink));

  await loza.notice('notice event', loza.string('family', 'logs'));
  await loza.track('checkout.page_view', loza.string('page', '/checkout'));
  await loza.audit('user.login', loza.userId('u_123'));
  await loza.security('auth.failure', loza.errorCode('AUTH_BAD_PASSWORD'));
  await loza.metric('latency', 42, loza.string('unit', 'ms'));
  await loza.count('requests', 4);
  await loza.gauge('cpu', 0.72);
  await loza.histogram('payload.bytes', 512);
  await loza.breadcrumb('nav.click', loza.string('button', 'submit'));
  await loza.flush();

  assert.ok(sink.getLength() >= 9);
});
